airflow/providers/cncf/kubernetes/provider.yaml

```diff
@@ -97,13 +97,13 @@ dependencies:
   # limiting minimum airflow version supported in cncf.kubernetes provider, due to the
   # potential breaking changes in Airflow Core as well (kubernetes is added as extra, so Airflow
   # core is not hard-limited via install-requirements, only by extra).
-  - kubernetes>=28.1.0,<=29.0.0
+  - kubernetes>=29.0.0,<=30.1.0
```

Comment:
Can we do that?
According to
https://github.com/kubernetes-client/python?tab=readme-ov-file#compatibility-matrix-of-supported-client-versions

> client 29.y.z: Kubernetes 1.28 or below (+-), Kubernetes 1.29 (✓), Kubernetes 1.30 or above (+-)

While we promise support for 1.26, 1.27, 1.28, 1.29

Reply:
This is the kubernetes client library version, not the k8s cluster version. Generally speaking any newer version of the library can talk to old versions of clusters as well (and we test it actually - part of this PR were all the k8s cluster tests with all supported versions.

Reply:
Kubernetes 1.29 or below (+-):  "However, everything they have in common (i.e., most APIs) will work":


```
+ client-python has features or API objects that may not be present in the Kubernetes cluster, either due to that client-python has additional new API, or that the server has removed old API. However, everything they have in common (i.e., most APIs) will work. Please note that alpha APIs may vanish or change significantly in a single release.
- The Kubernetes cluster has features the client-python library can't use, either due to the server has additional new API, or that client-python has removed old API. However, everything they share in common (i.e., most APIs) will work.
See the [CHANGELOG](https://github.com/kubernetes-client/python/blob/master/CHANGELOG.md) for a detailed description of cha
```

Reply:
(Or below) :D

Reply:
![image](https://github.com/apache/airflow/assets/595491/45370797-c148-439d-91fd-ebcf3336a94c)

Reply:
Also one more comment - here - this is the "cncf.kubernetes" provider dependency, that's one of the nice features of providers - that in case of any incompatibility (or when for whatever reason someone will want to use earlier version of the client, they will be able to use previous provider versions. Surely they will not have latest fixes, but they generally "work", so we are not really "limiting" anyone here - we just make latest cncf.kubernetes provider require the 1.29/1.30 library.